### PR TITLE
alsa-tools-1.0.29: add initial package

### PR DESCRIPTION
**Pull request by fps**
…nut cases. Maybe someone more experienced can take a look for the remaining three packages (see the comment in the patchPhase for the missing ones).

### CODE REVIEW

**Comment by peti** [comment]
The packages that are active in the current build compile and work correctly?

**Comment by fps** [comment]
Hi, these tools are rather specific for special soundcards. I packaged this mostly for having envy24control available for my m-audio delta66. I cannot confirm that the other bundled applications work correctly due to lack of having that specific hardware. envy24control works just fine from the result/ folder using nix-build. The rest builds fine except for the few nutcases :)

<pre>
fps@mango 12:19:30 ~/src/bld/nixpkgs (alsa-tools) $ find result/                                                                        
result/                                                                                                                                 
result/share                                                                                                                            
result/share/applications                                                                                                               
result/share/applications/hdspmixer.desktop                                                                                             
result/share/applications/hdspconf.desktop                                                                                              
result/share/aclocal                                                                                                                    
result/share/aclocal/ld10k1.m4                                                                                                          
result/share/pixmaps                                                                                                                    
result/share/pixmaps/hdspconf.png                                                                                                       
result/share/pixmaps/hdspmixer.png                                                                                                      
result/share/ld10k1                                                                                                                     
result/share/ld10k1/effects                                                                                                             
result/share/ld10k1/effects/switch_6.emu10k1                                                                                            
result/share/ld10k1/effects/prologic.emu10k1                                                                                            
result/share/ld10k1/effects/mono_switch_2x2.emu10k1                                                                                     
result/share/ld10k1/effects/mono_switch_2.emu10k1                                                                                       
result/share/ld10k1/effects/simple.emu10k1                                                                                              
result/share/ld10k1/effects/vol_master.emu10k1                                                                                          
result/share/ld10k1/effects/eq10.emu10k1                                                                                                
result/share/ld10k1/effects/switch_2.emu10k1                                                                                            
result/share/ld10k1/effects/tone.emu10k1                                                                                                
result/share/ld10k1/effects/sto51.emu10k1                                                                                               
result/share/ld10k1/effects/fxbus.emu10k1                                                                                               
result/share/ld10k1/effects/copy_2.emu10k1                                                                                              
result/share/ld10k1/effects/output.emu10k1                                                                                              
result/share/ld10k1/effects/vol_2.emu10k1                                                                                               
result/share/ld10k1/effects/switch_2x2.emu10k1                                                                                          
result/share/man                                                                                                                        
result/share/man/man1                                                                                                                   
result/share/man/man1/envy24control.1.gz                                                                                                
result/share/man/man1/cspctl.1.gz                                                                                                       
result/bin                                                                                                                              
result/bin/cspctl                                                                                                                       
result/bin/dl10k1                                                                                                                       
result/bin/hdsploader                                                                                                                   
result/bin/vxloader                                                                                                                     
result/bin/hdajacksensetest                                                                                                             
result/bin/as10k1                                                                                                                       
result/bin/hda-verb                                                                                                                     
result/bin/ld10k1d                                                                                                                      
result/bin/init_live                                                                                                                    
result/bin/envy24control                                                                                                                
result/bin/mixartloader                                                                                                                 
result/bin/hwmixvolume                                                                                                                  
result/bin/hdspmixer                                                                                                                    
result/bin/lo10k1                                                                                                                       
result/bin/pcxhrloader                                                                                                                  
result/bin/ld10k1                                                                                                                       
result/bin/hdspconf                                                                                                                     
result/bin/echomixer                                                                                                                    
result/bin/init_audigy_eq10                                                                                                             
result/bin/rmedigicontrol                                                                                                               
result/bin/us428control                                                                                                                 
result/bin/hdajackretask                                                                                                                
result/bin/init_audigy                                                                                                                  
result/bin/sscape_ctl
result/lib
result/lib/liblo10k1.so.0.0.0
result/lib/liblo10k1.la
result/lib/liblo10k1.so
result/lib/liblo10k1.so.0
result/sbin
result/include
result/include/lo10k1
result/include/lo10k1/ld10k1_error.h
result/include/lo10k1/ld10k1_fnc.h
result/include/lo10k1/lo10k1.h
result/include/lo10k1/version.h
result/include/lo10k1/liblo10k1lf.h
result/include/lo10k1/liblo10k1.h
result/include/lo10k1/comm.h
result/include/lo10k1/liblo10k1ef.h
</pre>

**Comment by fps** [comment]
And I took the nutcases out of the build. So yes, the remaining ones compile fine, but I cannot confirm them working correctly.

File: pkgs/os-specific/linux/alsa-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    homepage = http://www.alsa-project.org/;
+    description = "ALSA, the Advanced Linux Sound Architecture tools";
+
+    longDescription = ''
+      The Advanced Linux Sound Architecture (ALSA) provides audio and
+      MIDI functionality to the Linux-based operating system.
+    '';
+
+    platforms = stdenv.lib.platforms.linux;
+    maintainers = [ "Florian Paul Schmidt <mista.tapas@gmx.net>" ];
```
Comment (pSub) [inline pkgs/os-specific/linux/alsa-tools/default.nix]: Please add yourself to `lib/maintainers.nix` and use here the identifier.

File: pkgs/os-specific/linux/alsa-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    urls = [
+      "ftp://ftp.alsa-project.org/pub/tools/${name}.tar.bz2"
+      "http://alsa.cybermirror.org/tools/${name}.tar.bz2"
+    ];
+    sha256 = "1lgvyb81md25s9ciswpdsbibmx9s030kvyylf0673w3kbamz1awl";
+  };
+
+  enableParallelBuilding = true;
+
+  phases = "unpackPhase patchPhase configurePhase buildPhase installPhase fixupPhase";
```
Comment (pSub) [inline pkgs/os-specific/linux/alsa-tools/default.nix]: You don't have to list those phases explicitly.

**Comment by fps** [comment]
Also a question remains on my part: The <code>enableParallelBuilding = true;</code> seems to have no effect. I guess it's because I manually hijacked some phases? Should I take it out?

**Comment by pSub** [comment]
You're right, that is because you have overwritten `buildPhase`. If you know that this project can be build in parallel, take a look at the default build phase.

**Comment by fps** [comment]
I'm not sure about it and the package is small enough that the time saving is probably not worth investigating it deeply. I'll take it out for now..

**Comment by peti** [comment]
If the build succeeds and produces useful results, it's fine IMHO to merge it now, even if some of the tools don't work yet.

**Comment by fps** [comment]
Excuse my ignorance, since I'm pretty much still a complete nixos-noob. What other precautions besides nix-build'ing it (and running the tool that I can from the result/ folder) should I take to make sure it doesn't mess up anything?

**Comment by peti** [comment]
I don't believe that a newly added package can mess up things much. A patch that modifies, say `glibc`, would need more attention, but this doesn't seem to be the case here.

**Comment by fps** [comment]
OK, thanks for your help in getting this merged. Appreciated :)

**Pull request merged**